edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3008,26 +3012,28 @@ public Object doInHibernate(final Session session) throws HibernateException {
 			}
 		}
 
-		return calculateCategoryScore(studentUuid, category.getId(), gradeRecords, includeNonReleasedItems);
+		return calculateCategoryScore(studentUuid, category.getId(), gradeRecords, includeNonReleasedItems, gb.getCategory_type());
 	}
 
 	@Override
-	public Optional<CategoryScoreData> calculateCategoryScore(final Long gradebookId, final String studentUuid, final Long categoryId, final boolean includeNonReleasedItems) {
+	public Optional<CategoryScoreData> calculateCategoryScore(final Object gradebook, final String studentUuid, final Long categoryId, final boolean includeNonReleasedItems) {
+
+		final Gradebook gb = (Gradebook) gradebook;
```

Comment:
How about just passing in int categoryType instead of passing the whole gradebook object and using this cast?

Reply:
That was my first thought, but it also needs the gradebook id so I thought it might be better to pass the object all things considered.

Reply:
Given that only two pieces of information from the Gradebook object are needed, and one of those is already being supplied, I would prefer to keep at least some typing in place and just add the additional int parameter. But that's just my opinion.

Reply:
Agree with Paul here